### ISSUE TITLE
CompileFlags on source doesnt change inferred flags for headers

### ISSUE THREAD

**Issue by kiwixz** [clangd/clangd#1078]
The simplest is to describe my use case: I have some cuda (*.cu) files with nvcc flags that clangd doesnt understand.

So I added a fragment to my .clangd to remove those flags from cuda files:
```
If:
  PathMatch: ".*\\.cu"

CompileFlags:
  Remove: ...
```

While it now tries to parse the cuda files, the problem is that they include some headers (*.h) files which seems to still pick the removed flags.

I think that the flags should be removed from the guess clangd has to do for header files, if they were removed from the source file.

**Comment by HighCommander4** [comment]
I agree that this sounds useful, but as a workaround for this specific case, is there any harm in removing the flags that clangd doesn't understand from all files?

**Comment by kiwixz** [comment]
Actually this is worse than I first thought, even without the `If` I could not remove the flags via `.clangd`.  I actually dont think the `CompileFlags` option apply to headers in any way.

**Comment by sam-mccall** [comment]
Is the .clangd file above the header files in the directory structure?

You should be able to use it to change the flags used to parse headers, but it's hard to say if you're seeing a bug without more details.

**Comment by kiwixz** [comment]
Sorry, I was mistaken in my previous comment.  `CompileFlags` works has expected.

I think the original feature request still stands, and would be useful to i.e. change the standard version for a handful of files.
But a little bit of regex-fu on the paths may suffice to isolate the desired header files in most cases.

**Comment by kiwixz** [comment]
Another use case maybe more convincing: opening a system header.  The header will inherit the flags of the source file, but will not apply even a project-scope global `CompileFlags`.
